.github/workflows/secondary.yml

```diff
@@ -0,0 +1,40 @@
+name: artifacts
+
+on: [push, pull_request]
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - uses: actions/checkout@v2
+    - id: set-matrix
+      run: echo "::set-output name=matrix::$(ls configs/katello/ | sed '/-/!{s/$/_/}' | sort -rV | sed 's/_$//' | grep yaml | grep -v nightly | head -n 3 | jq -cRs 'split("\n")[:-1]')"
```

Comment:
I'm not sure what you're exactly after, but how about using this as a base?
```console
$ ls configs/katello | xargs -I '{}' basename '{}' .yaml | sort -n | tail -n 3
3.8
3.9
4.0
```

Reply:
Or even better: use `basename --suffix=.yaml configs/katello/*.yaml | sort -n | tail -n 3`

Reply:
Ah, ok yes I prefer a more compact expression - but in this case I just wanted the 3 most recent versions, "4.0, 3.18, 3.17", for example. The default sorting doesn't get me there.

Reply:
I thought that's what `sort -n` does. From `man sort`:

```
-n, --numeric-sort
              compare according to string numerical value
```

Looks like we should use `-V`:
```
-V, --version-sort
              natural sort of (version) numbers within text
```

So then it becomes:
```console
$ basename --suffix=.yaml configs/katello/*.yaml | grep -v nightly | sort -V | tail -n 3
3.17
3.18
4.0
```

Reply:
And I now see you already found that.